state-chain/pallets/cf-broadcast/src/lib.rs

```diff
@@ -409,9 +409,11 @@ pub mod pallet {
 						Self::deposit_event(Event::<T, I>::BroadcastAttemptTimeout {
 							broadcast_attempt_id: *attempt_id,
 						});
-						if let Some(broadcast_attempt) = Self::take_awaiting_broadcast(*attempt_id)
+						if let Some((broadcast_attempt, nominee)) =
+							Self::take_awaiting_broadcast(*attempt_id)
 						{
-							Self::start_next_broadcast_attempt(broadcast_attempt);
+							FailedBroadcasters::<T, I>::append(attempt_id.broadcast_id, nominee);
```

Comment:
I think we should use the `transaction_signing_failure` logic here, that way if everyone is attempted and everyone times out we abort. 

Currently, what happens here, is if the last broadcast attempt fails due to a timeout, adding the final authority to failed broadcasters, we fill up FailedBroadasters storage... then we can't nominate anyone, and we enter an infinite retry loop, a retry every block trying to nominate someone, but everyone's listed as Failed. 

As part of this PR, I think we should probably look at using `FailedBroadcasters.len()` instead of `attempt_count` for 
```rust
if signing_attempt.broadcast_attempt.broadcast_attempt_id.attempt_count >=
				T::EpochInfo::current_authority_count()
					.checked_sub(1)
					.expect("We must have at least one authority")
```

which is currently in the `transaction_signing_failure` logic, but eventually will be shared.

We should also add a test for this case, what happens when the *last* authority fails, and ensure that it gets aborted.